File: Data Structures & Algorithms/group-shifted-strings/submission-0.js
class Solution {
    /**
     * @param {string[]} strings
     * @return {string[][]}
     */
    groupStrings(strings) {
        function getHash(str) {
            const key = []

            for (let i = 0; i < str.length - 1; i++) {
                const diff = str.charCodeAt(i + 1) - str.charCodeAt(i)
                const round = (diff + 26) % 26
                key.push(String.fromCharCode(round + 'a'.charCodeAt(0)))
            }
            
            const res = key.join('')
            return res.length > 0 ? res : '(empty)'
        }

        const m = {}
        
        for (let str of strings) {
            const key = getHash(str)
            if (!m[key]) {
                m[key] = []
            }
            m[key].push(str)
        }

        console.log(m)

        return Object.values(m)
    }
}

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        def hammingWeight(a: int) -> int:
            res = 0
            while a:
                a &= (a-1)
                res += 1
            
            return res

        return [hammingWeight(i) for i in range(n + 1)]
        
        